controller/controller.go

```diff
@@ -182,10 +182,19 @@ func (c *notificationController) processApp(app *unstructured.Unstructured, logE
 			logEntry.Debugf("Failed to execute condition of trigger %s: %v", triggerKey, err)
 		}
 		recipients := c.getRecipients(app, triggerKey)
-		triggerAnnotation := fmt.Sprintf("%s.%s", triggerKey, sharedrecipients.AnnotationPostfix)
 		logEntry.Infof("Trigger %s result: %v", triggerKey, triggered)
 		c.metricsRegistry.IncTriggerEvaluationsCounter(triggerKey, triggered)
-		if triggered {
+		if !triggered {
+			for recipient := range recipients {
+				triggerAnnotation := fmt.Sprintf("%s.%s.%s", triggerKey, strings.Replace(recipient, ":", ".", 1), sharedrecipients.AnnotationPostfix)
```

Comment:
nit: can you please move `fmt.Sprintf("%s.%s.%s", triggerKey, strings.Replace(recipient, ":", ".", 1), sharedrecipients.AnnotationPostfix)` into function?

Reply:
Thank you, done.